### PR TITLE
To allow multiple join conditions when performing inner join

### PR DESCRIPTION
**Pull request by TerenceHinrichsen**
Added a clause to the Join discriminated union, although I think this could possibly be a drop-in replacement for the current inner join, i.e you can use the `InnerJoinOnMany` even when only joining on 1 column.

However, I do not want to break any of the existing code, it is probably better as an augmentation of the current functionality?

Usage example: 
```
  let selectQuery =
    select {
      schema "mySchema"
      table "Event"
      innerJoinOnMany "mySchema.History" ["Guid","Event.Guid"; "Version","Event.Version"]
  }
```

Disclaimer: This is my first ever submission to any open source project, so please tell me if I am being a complete idiot and missing something obvious.

### CODE REVIEW

File: src/Dapper.FSharp/MSSQL.fs [hunk 1]
```diff
@@ -168,10 +175,10 @@ module private Evaluators =
 
 [<AbstractClass;Sealed>]
 type Deconstructor =
-    static member select<'a> (q:SelectQuery) = q |> GenericDeconstructor.select1<'a> Evaluators.evalSelectQuery
-    static member select<'a,'b> (q:SelectQuery) = q |> GenericDeconstructor.select2<'a,'b> Evaluators.evalSelectQuery
+    static member select<'a> (q:SelectQuery)       = q |> GenericDeconstructor.select1<'a> Evaluators.evalSelectQuery
```
Comment (Dzoukr) [inline src/Dapper.FSharp/MSSQL.fs:178]: Please, don't align the code like Excel. :) 
See: https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting
and: https://caml.inria.fr/resources/doc/guides/guidelines.en.html

**Comment by Dzoukr** [comment]
Hi Terence, thanks for the contribution. It looks good - can you please fix those formatting issues and add some tests to see it works?

**Comment by TerenceHinrichsen** [comment]
Hi Roman, I have created a test, but I am unable to run it without the local.settings.json file.. can you tell me what the required keys for the file is so I can create it? Or how should I be running the tests?

**Comment by Dzoukr** [comment]
Oh, my fault, I should add this to docs:

```
{
  "mssqlConnectionString": "********",  
  "mysqlConnectionString": "********",  
  "postgresConnectionString": "********" 
}
```

I use docker images just for this purpose (no need to install it locally)

**Comment by TerenceHinrichsen** [comment]
Morning Roman, 
I have added a tests for the new innerjoin type. I could get it running the MSSQL tests and that test passed. 
Would you mind reviewing the tests before I add more?

**Comment by Dzoukr** [comment]
Hi Terence, it looks good now!

Looking at the new keyword `innerJoinOnMany` (and your initial comment), we could maybe take this as the new left/inner join behavior and instead of adding a brand new keyword, we could allow to overload the `leftJoin`, `innerJoin` keywords (it's now possible in F# 5 - see https://docs.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-50#preview-overloads-of-custom-keywords-in-computation-expressions) because as you mentioned at the very beginning: what's the difference between current join and yours with a single value in the list?

So please finish the tests and I can merge it and reuse your amazing work to make it even cooler, or (if you like) you can change the structure of join directly and plug it together. But no pressure - you've done enough already and I really appreciate your contribution, so if you just want to finish it, merge it and let me do the F#5 stuff, it's 100% ok! 👍 

Anyway, what do you think about it?

**Comment by TerenceHinrichsen** [comment]
Awesome! Thank you very much, I will read up about the overloading and give it a bash (eager to learn). Will let you know how it goes! Thank you SO much for your help and friendly feedback!

I must say, that we use this library extensively at work and I have implemented a personal "pet project" with it as well as it is soo simple and easy. Looking forward to contributing to it and making it even more amazing!

**Comment by Dzoukr** [comment]
I thank YOU, Terence! Every contribution like this will make this tiny lib even better! 👍 

Btw... watchout for `global.json` - you would need to change the SDK version to be allowed to use F#5 features. ;)

**Comment by TerenceHinrichsen** [comment]
For #36 
I am sorry it took > 4 months (life happened).
Was able to add the overload on Innerjoin, so there is not a "innerjoinonmany" keyword added to the CE. I created tests and on MSSQL it seems to be working, I have tried to get the MySQL and PostGreSQL up and running, but my experience with docker is seriously lacking - I will be working on fixing that soon. I hope that this pull request attempt is in a better position than my previous attempt.
As always, I await your feedback @Dzoukr !

**Comment by JordanMarr** [comment]
Sorry if my changes caused you any merge headaches. 😁

**Comment by TerenceHinrichsen** [comment]
I would submit to that amazing addition any day of the week, my biggest hurdle was getting time to get tests done 🤨

**Comment by Dzoukr** [comment]
Thanks a lot! I'll look at the tests for other providers and hopefully release soon!

**Comment by Dzoukr** [comment]
Ufff, took me a while to fix a PostgreSQL bug + add few overload for already existing _*Many_ methods, but it's done and will be release in a few minutes.

**Comment by Dzoukr** [comment]
And FYI I had to change a test - comparing two DateTimes via _Expect.isEqual_ fails, even if they differ by a few ticks. Probably the roundtrip from type -> db -> type doesn't make it _exactly_ the same value. 🤷‍♂️

**Comment by Dzoukr** [comment]
Released as v2.2.0 - thanks again, man!!! 🎉

**Pull request merged**